src/sklearn_raster/features.py

```diff
@@ -196,25 +207,31 @@ def _postprocess_ufunc_output(
 class DataArrayFeatures(FeatureArray):
     """Features stored in an xarray DataArray of shape (features, ...)."""
 
-    def __init__(self, features: xr.DataArray, nodata_input: NoDataType = None):
+    def __init__(
+        self,
+        features: xr.DataArray,
+        nodata_input: NoDataType | MissingType = MissingType.MISSING,
+    ):
         super().__init__(features, nodata_input=nodata_input)
         self.feature_dim_name = features.dims[self.feature_dim]
 
     @property
     def feature_names(self) -> NDArray:
         return self.feature_array[self.feature_dim_name].values
 
-    def _validate_nodata_input(self, nodata_input: NoDataType) -> NDArray:
+    def _validate_nodata_input(self, nodata_input: NoDataType | MissingType) -> NDArray:
         """
         Get an array of NoData values in the shape (features,) based on user input and
         DataArray metadata.
         """
-        # Defer to user-provided NoData values over stored attributes
-        if nodata_input is not None:
-            return super()._validate_nodata_input(nodata_input)
+        # Infer NoData from _FillValue if present (or None) for all features
+        if nodata_input is MissingType.MISSING:
+            return np.full(
+                (self.n_features,), self.feature_array.attrs.get("_FillValue")
+            )
```

Comment:
I flipped the order of the conditional here (and below), just to avoid having to think through the double negative of NoData not being missing